### DISCUSSION TITLE
Customise the Status segment

### DISCUSSION THREAD

**Discussion by ccjmne** [JanDeDobbeleer/oh-my-posh#5317] [Troubleshoot] [answered]
Hey,

I'd love to customise my `status` segment, to show things like `SIGKILL`, `SIGINT`, `SIGTSTP`, `SIGPIPE|0|0|123`...

Basically, I want to have this output:
```shell
kill -l "${pipestatus[@]}" | paste -sd'|'
```
... but I cannot figure out how to either:
- access the `$pipestatus` in a `command` segment (I tried with the `set_poshcontext()`, too, to no avail), or
- leverage the `status` segment's `.Code` template property (is that the only way I can refer to the previous command's status code from within oh-my-posh?) in my `command` segment.

Does that sound at all possible? If it currently isn't, should it be the object of a Feature Request?

Thanks in advance!

**Comment by lewis-yeung** [comment]
Does [this][docs] fit your use case?

> In case you want the reason for the exit code instead of code itself, you can use the `reason` function:
> 
> ```
> {{ if eq .Code 0 }}\uf00c{{ else }}\uf071 {{ reason .Code }}{{ end }}
> ```

It shows something like `SIGKILL`, `SIGINT`, `SIGTSTP`, etc., for a non-zero status code.

[docs]: https://ohmyposh.dev/docs/segments/system/status#status-template

**Reply by ccjmne** [reply]
Thanks, @lewis-yeung!
That's pretty neat, but doesn't quite work for pipe-chained calls, however.

Consider `yes | head -10 | awk '{ invalid'`, I'd just get `ERROR` printed out, while it'd be nice to have `PIPE|0|1`.

Even more so when somebody pipes a failing command into `tee`: I like my `false | tee log.txt` to let me know that the outcome was `1|0` rather than simply "all good"!

**Reply by JanDeDobbeleer** [reply]
You also use `.String` to get the `pipestatus` separated by `|`.

**Reply by ccjmne** [reply]
Oooh, I see. I could just use a template, and do my manipulation there:
With `status_template` as `{{ reason .Code }}`... and removing the "SIG" prefix can be done inside the template as well, with Sprig. Alright, that works! Thank you :)

**Reply by ccjmne** [reply, marked as answer]
For the others who might want it:
A `status` segment whose `status_template` property is:
```
{{ if eq 1 .Code }}1{{ else }}{{ trimPrefix "SIG" (reason .Code) }}{{ end }}'
```
... appears to yield the equivalent to `kill -l "${pipestatus[@]}" | paste -sd"|"`.
